src/native/libs/System.Security.Cryptography.Native/pal_evp.c

```diff
@@ -139,6 +168,36 @@ int32_t CryptoNative_EvpDigestOneShot(const EVP_MD* type, const void* source, in
     return ret;
 }
 
+int32_t CryptoNative_EvpDigestXOFOneShot(const EVP_MD* type, const void* source, int32_t sourceSize, uint8_t* md, uint32_t len)
+{
+    ERR_clear_error();
+
+    if (type == NULL || sourceSize < 0 || md == NULL)
+    {
+        return 0;
+    }
+
+    EVP_MD_CTX* ctx = CryptoNative_EvpMdCtxCreate(type);
+
+    if (ctx == NULL)
+    {
+        return 0;
+    }
+
+    int32_t ret = EVP_DigestUpdate(ctx, source, (size_t)sourceSize);
+
+    if (ret != SUCCESS)
+    {
+        CryptoNative_EvpMdCtxDestroy(ctx);
+        return 0;
+    }
```

Comment:
On failure does the managed caller try to use GetLastError to retrieve failure information?  I'm wondering if the call to CryptoNative_EvpMdCtxDestroy will overwrite any error information from the EVP_DigestUpdate call.

Reply:
OpenSSL does this a bit differently than Win32. It maintains an error queue. Successful functions don't set the last error to "no error", they just don't add anything to the queue. So `EVP_DigestUpdate` error is still on the queue, then when we check the return value for failed we use ERR_peek_last_error to grab the error, then ERR_clear_error to clear the queue out.